lib/graphql/schema/field.rb

```diff
@@ -132,36 +144,6 @@ def argument_class(new_arg_class = nil)
           end
         end
       end
-
-      # Call the given block with the field ... do end DSL
-      def with_proxy_dsl(&block)
-        field_proxy = FieldProxy.new(self, argument_class: self.class.argument_class)
-        field_proxy.instance_eval(&block)
-      end
-
-      # This object exists only to be `instance_eval`'d
-      # when the `field(...)` method is called with a block.
-      # This object receives that block.
-      class FieldProxy
-        def initialize(field, argument_class:)
-          @field = field
-          @argument_class = argument_class
-        end
-
-        # This is the `argument(...)` DSL for class-based field definitons
-        def argument(*args)
-          arg_defn = @argument_class.new(*args)
-          @field.arguments[arg_defn.name] = arg_defn
-        end
-
-        def description(text)
-          if @field.description
-            fail "`description` was already assigned to this field (#{@field.name}). Assign it as a keyword argument or in the block, but not both."
```

Comment:
Makes sense to remove this, since previously, duplication was _only_ possible with `description`, but now it's possible with lots of things, and we can't really guard against all of them.